### PR TITLE
Quests/player module update

### LINKED ISSUE #1163: Incorporate quests in the player struct
Quests are only partially implemented in chiventure - they are present in the game struct but cannot be stored in the player struct. It'd be great if the player could store a hash table of quests. See the [current implementation wiki](https://github.com/uchicago-cs/chiventure/wiki/Quests-~-Current-Implementation) for more details.

### PR DESCRIPTION
**Pull request by Castel-Jaloux**
Updated the player module (player.h and player.c) within the game-state directory in order to properly integrate quests with the player module. Due to circular dependency issues, we cannot directly import quests into the player module. To resolve this issue, we created player_quest and player_task structs to be included within the player struct, as well as corresponding functions (new, init, free) to ensure that they are properly functional.

### CODE REVIEW

**Comment by elizabethli31** [comment]
Add asserts for player_quest_hash_free and player_task_hash_free

**Review by elizabethli31 — APPROVED** [review]
Everything looks good and consistent. The only comment I had before was to add asserts for the hash table functions, so that it wasn't automatically returning success, and this was fixed in the last commit.

File: tests/game-state/test_player.c [hunk 1]
```diff
@@ -13,6 +13,66 @@
 #include "battle/battle_default_objects.h"
 #include "battle/battle_structs.h"
 
+Test(player, quest_new)
```
Comment (philipchenaux) [inline tests/game-state/test_player.c:17]: It is also good to include function comments to document what each of your tests actually test.

File: src/game-state/src/player.c [hunk 2]
```diff
@@ -125,11 +127,109 @@ int player_free(player_t* player)
...
+    {
+        player_task_hash_free(player->player_tasks);
+    }
+
     free(player);
     
     return SUCCESS;
 }
 
+player_quest_t *player_quest_new(char *quest_id, int completion)
```
Comment (philipchenaux) [inline src/game-state/src/player.c:146]: Very minor stylistic comment, but note that you should try be consistent with the rest of the functions and include a /* See player.h */ function comment.

**Review by philipchenaux — COMMENTED** [review]
A few minor comments that can quickly be resolved, but besides that the actual code base looks good!

File: src/game-state/src/player.c (hunk 2 above)
Comment (JSerf02) [reply src/game-state/src/player.c:146]: Just fixed

**Review by philipchenaux — APPROVED** [review]
Looks good to merge (LGTM)! Great work.

**Comment by philipchenaux** [comment]
Issue Score: Excellent

Comments:
Great job on this PR both to those that created it and reviewed it!

**Pull request merged**